web-api/src/storage.rs

```diff
@@ -93,15 +72,14 @@ pub(crate) trait Document {
         params: KnnSearchParams<'a, impl IntoIterator<Item = &'a DocumentId>>,
     ) -> Result<Vec<PersonalizedDocument>, Error>;
 
-    async fn insert(
-        &self,
-        documents: Vec<(IngestedDocument, NormalizedEmbedding)>,
-    ) -> Result<(), InsertionError>;
+    /// Inserts the documents and reports failed ids.
+    async fn insert(&self, documents: Vec<IngestedDocument>) -> Result<Vec<DocumentId>, Error>;
```

Comment:
:wrench:  Documents failing is an error. Why do we return it now as a non error? Also why return it as a vec which looks like it could e.g. be the list of actually inserted documents or something like that?